packages/knip/src/typescript/visitors/exports/exportDeclaration.ts

```diff
@@ -15,15 +15,27 @@ export default visit(
         const type = node.isTypeOnly ? SymbolType.TYPE : SymbolType.UNKNOWN;
         const sourceFile: BoundSourceFile = node.getSourceFile();
         const declarations = sourceFile.getNamedDeclarations?.();
-        return node.exportClause.elements.map(element => {
+        const allElementStrings = node.exportClause.elements.map(element => element.getFullText());
+        return node.exportClause.elements.map((element, index, arr) => {
           const identifier = String(element.name.text);
           const propName = element.propertyName?.text;
           // @ts-expect-error TODO Fix (convenience in addExport)
           // const symbol = element.symbol ?? declarations?.get(identifier)?.find((d: ts.Node) => d !== element)?.symbol;
           const symbol = declarations?.get(propName ?? identifier)?.[0]?.symbol;
           const pos = element.name.pos;
-          const fix: Fix = isFixExports || isFixTypes ? [element.getStart(), element.getEnd()] : undefined;
-          return { node: element, symbol, identifier, type, pos, fix };
+
+          const exportClause = node.exportClause;
+          const fix1: Fix =
+            isFixExports || isFixTypes
+              ? {
+                  element: element.getFullText(),
+                  pos: [exportClause!.getStart(), exportClause!.getEnd()], //  TODO wrong, must be coordinates of {}
+                  allElements: allElementStrings,
```

Comment:
Re. `allElementStrings`: the idea is to do the necessary work for the `IssueFixer` only if `isFixExports || isFixTypes` is true.

The bigger question is: do we want to calculate and store things upfront (like here) or do the heavy lifting later on in the `IssueFixer` (like the parser I did).

Here in AST area it's easier and more accurate. But we are probably doing a lot more work than necessary, because we're doing it for each export even though we may not remove it (usually a relatively small percentage is unused). That said, performance isn't the most important thing in `--fix` mode.

In a custom parser "after the fact" we're doing the work efficiently, but it's a bit clumsy. As it stands it also seems a bit less intrusive or safer with regards to comments and such.